### PR TITLE
python3Packages.pydantic-core: init at 2.1.3

### PR DESCRIPTION
**Pull request by blaggacao**
###### Description of changes

This package provides the core functionality for pydantic V2 validation and serialization.
Pydantic-core is currently around 17x faster than pydantic V1.

When this package is reviewed, attested and merged, we can start considering to bump pydantic to V2.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/typing-extensions/default.nix
Comment (blaggacao) [inline pkgs/development/python-modules/typing-extensions/default.nix:1]: https://github.com/pydantic/pydantic-core/blob/main/pyproject.toml#L36

File: pkgs/development/python-modules/pydantic-core/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+, fetchFromGitHub
+, cargo
+, rustPlatform
+, rustc
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "pydantic-core";
+  version = "2.1.3";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pydantic-core/default.nix:12]: The commit for this contains an extra v in the version number

File: pkgs/development/python-modules/typing-extensions/default.nix
Comment (SuperSandro2000) [reply pkgs/development/python-modules/typing-extensions/default.nix:1]: Can we just relax this if this is causing to many rebuilds?

Comment (blaggacao) [reply pkgs/development/python-modules/typing-extensions/default.nix:1]: I thin we can: https://github.com/pydantic/pydantic-core/pull/728

How much do we need to care about PyPy `3.{7,8}`?

Comment (blaggacao) [reply pkgs/development/python-modules/typing-extensions/default.nix:1]: Local `nixpkgs-review` failed on pythonRelaxDepsHook`.... Need to investigate more.

File: pkgs/development/python-modules/pydantic-core/default.nix (hunk 1 above)
Comment (blaggacao) [reply pkgs/development/python-modules/pydantic-core/default.nix:13]: fixed, thx

**Comment by blaggacao** [comment]
I'd temporarily put this on hold for further local testing and wiring of `pytest` hooks and reopen when I've successfully completed those tests.

File: pkgs/build-support/rust/hooks/maturin-build-hook.sh [hunk 2]
```diff
@@ -34,6 +32,9 @@ maturinBuildHook() {
     mkdir -p dist
     mv target/wheels/*.whl dist/
 
+    # These are python build hooks and may depend on ./dist
+    runHook postBuild
+
```
Comment (blaggacao) [inline pkgs/build-support/rust/hooks/maturin-build-hook.sh:37]: One example: [python-relax-deps-hook](https://github.com/NixOS/nixpkgs/blob/9e17672672a9906a9f313601c6301d4ac7f95580/pkgs/development/interpreters/python/hooks/python-relax-deps-hook.sh#L78), which is a  `postBuild` hook.

**Comment by blaggacao** [comment]
:hourglass: I'm currently plowing through 1823 rebuilds on my machine and will need to supply (as of now) at least 2 fixes to `maturinBuildHook`-affected packages with (what appears to be) new failures. If you happen to review, please keep this (still) long running local work in mind.

**Pull request closed without merging**